source/fem-dem/cfd_dem_coupling.cc

```diff
@@ -1246,22 +1266,53 @@ void
 CFDDEMSolver<dim>::print_particles_summary()
 {
   // Write particle Velocity
-  for (auto &particle : this->particle_handler)
+  std::map<int, Particles::ParticleIterator<dim>> global_particles;
+  unsigned int current_id, current_id_max = 0;
+
+  // Mapping of all particles & find the max id on current processor
+  for (auto particle = this->particle_handler.begin();
+       particle != this->particle_handler.end();
+       ++particle)
     {
-      auto particle_properties = particle.get_properties();
-      this->pcout << "Particle Summary" << std::endl;
+      current_id     = particle->get_id();
+      current_id_max = std::max(current_id, current_id_max);
 
-      std::stringstream ss;
+      global_particles.insert({current_id, particle});
+    }
 
-      ss << std::setprecision(6) << "id: " << particle.get_id() << ",  "
-         << "x: " << particle.get_location()[0] << ",  "
-         << "y: " << particle.get_location()[1] << ",  "
-         << "z: " << particle.get_location()[2] << ",  "
-         << "v_x: " << particle_properties[DEM::PropertiesIndex::v_x] << ",  "
-         << "v_y: " << particle_properties[DEM::PropertiesIndex::v_y] << ",  "
-         << "vz: " << particle_properties[DEM::PropertiesIndex::v_z];
+  // Find global max particle index
+  unsigned int id_max =
+    Utilities::MPI::max(current_id_max, this->mpi_communicator);
 
-      announce_string(this->pcout, ss.str(), '-');
+  // Print particle info one by one in ascending order
+  for (unsigned int i = 0; i <= id_max; i++)
+    {
+      for (auto &iterator : global_particles)
+        {
+          unsigned int id = iterator.first;
+          if (id == i)
+            {
+              auto particle            = iterator.second;
+              auto particle_properties = particle->get_properties();
+
+              this->pcout << "Particle Summary" << std::endl;
+
+              std::stringstream ss;
+              ss << std::setprecision(6) << "id: " << id << ",  "
+                 << "x: " << particle->get_location()[0] << ",  "
+                 << "y: " << particle->get_location()[1] << ",  "
+                 << "z: " << particle->get_location()[2] << ",  "
+                 << "v_x: " << particle_properties[DEM::PropertiesIndex::v_x]
+                 << ",  "
+                 << "v_y: " << particle_properties[DEM::PropertiesIndex::v_y]
+                 << ",  "
+                 << "vz: " << particle_properties[DEM::PropertiesIndex::v_z];
+
+              announce_string(this->pcout, ss.str(), '-');
+            }
+        }
+      usleep(100);
+      MPI_Barrier(this->mpi_communicator);
```

Comment:
We could use the print_xyz in function, or print in the same manner (list and not 1 header per particle)